spec/integration/api/photos_controller_spec.rb

```diff
@@ -0,0 +1,293 @@
+      it "with correct access token" do
+        get(
+          api_v1_photos_path,
+          params: {access_token: access_token}
+        )
+        expect(response.status).to eq(200)
+        photos = JSON.parse(response.body)
+        expect(photos.length).to eq(2)
+      end
+    end
+
+    context "fails" do
+      it "with invalid access token" do
+        delete(
+          api_v1_photos_path,
+          params: {access_token: "999_999_999"}
+        )
+        expect(response.status).to eq(401)
+      end
+    end
+  end
+
+  describe "#create" do
+    before do
+      @encoded_photo = Rack::Test::UploadedFile.new(
+        Rails.root.join("spec", "fixtures", "button.png").to_s,
+        "image/png"
+      )
+    end
+
+    context "succeeds" do
+      it "with valid encoded file no arguments" do
+        post(
+          api_v1_photos_path,
+          params: {image: @encoded_photo, access_token: access_token}
+        )
+        expect(response.status).to eq(200)
+        photo = JSON.parse(response.body)
+        ref_photo = auth.user.photos.reload.find_by(guid: photo["guid"])
+        expect(ref_photo.pending).to be_falsey
+        confirm_photo_format(photo, ref_photo, auth.user)
+      end
+
+      it "with valid encoded file set as pending" do
+        post(
+          api_v1_photos_path,
+          params: {image: @encoded_photo, pending: false, access_token: access_token}
+        )
+        expect(response.status).to eq(200)
+        photo = JSON.parse(response.body)
+        expect(photo.has_key?("post")).to be_falsey
+        ref_photo = auth.user.photos.reload.find_by(guid: photo["guid"])
+        expect(ref_photo.pending).to be_falsey
+        confirm_photo_format(photo, ref_photo, auth.user)
+
+        post(
+          api_v1_photos_path,
+          params: {image: @encoded_photo, pending: true, access_token: access_token}
+        )
+        expect(response.status).to eq(200)
+        photo = JSON.parse(response.body)
+        ref_photo = auth.user.photos.reload.find_by(guid: photo["guid"])
+        expect(ref_photo.pending).to be_truthy
+      end
+
+      it "with valid encoded file as profile photo" do
+        original_profile_pic = auth.user.person.profile.image_url
+        post(
+          api_v1_photos_path,
+          params: {image: @encoded_photo, set_profile_photo: true, access_token: access_token}
+        )
+        expect(response.status).to eq(200)
+        expect(auth.user.reload.person.profile.image_url).not_to eq(original_profile_pic)
+      end
+    end
+
+    context "fails" do
+      it "with no image" do
+        post(
+          api_v1_photos_path,
+          params: {access_token: access_token}
+        )
+        expect(response.status).to eq(422)
+        expect(response.body).to eq(I18n.t("api.endpoint_errors.photos.failed_create"))
+      end
+
+      it "with non-image file" do
+        text_file = Rack::Test::UploadedFile.new(
+          Rails.root.join("AUTHORS").to_s,
```

Comment:
It may be better to use a fixture file or at least the README file.

Reply:
Yeah I just need a persistent file that's not an image, but I didn't see the point of generating one with all of the existing text files already in the code base.  README would work too.